upgrader.go

```diff
@@ -24,6 +25,8 @@ type Options struct {
 	UpgradeTimeout time.Duration
 	// The PID of a ready process is written to this file.
 	PIDFile string
+	// ListenConfig is a custom ListenConfig. Defaults to an empty ListenConfig
+	ListenConfig net.ListenConfig
```

Comment:
Please make this a *ListenConfig, then we can later switch to an interface if need be.

```suggestion
	ListenConfig *net.ListenConfig
```

Reply:
👍